### DISCUSSION TITLE
trusted-replace-outbound-text

### DISCUSSION THREAD

**Discussion by uBlock-user** [uBlockOrigin/uBlock-issues#3292] [General]
I'm testing this filter - `www.reddit.com##+js(trusted-replace-outbound-text, JSON.stringify, /^.*csrf_token.*$/, '{"source:global","screen":{"height":0,"width":0}","view_stats":{"comments":0}}')` as a fix for https://github.com/uBlockOrigin/uBlock-discussions/discussions/819#discussioncomment-7218569 and it does what I want it to do, ~~but server responds with 400 status(Internel server error), is it because this payload is not in JSON format ? If yes, then how to convert to JSON format ?~~ Internal Server error issue fixed with this version -- 

`www.reddit.com##+js(trusted-replace-outbound-text, JSON.stringify, /^.*csrf_token.*$/, '{"source": "global","screen": {"height": 480, "width": 840},"view_stats":{"comments": 100}}')`

But comments won't load still. How to fix that ?

@gorhill

**Comment by gorhill** [comment]
If I use `www.reddit.com##+js(trusted-replace-outbound-text, JSON.stringify, '/"csrf_token":.*?,/')`, the comment loads, so this means the server expect something in the payload to be there in order to respond properly.

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/f9825567-15f1-4b6b-b595-c9e3a3f6fd1c)

**Reply by uBlock-user** [reply]
Does this pattern remove the entire payload ?

**Reply by uBlock-user** [reply]
This is their script -- 

```
if (s?.ok) {
      "basic" === s.type && s.headers.has("x-verified-res") && ((e, t = hn) => {
         if (e.includes("global") && e.includes("view") && e.includes("screen")) {
            for (let n = 0; n < t.length; n++)
               if (!e.includes(t[n])) return !1;
            return !0
         }
         return null
      })(i) && (Ur || (Ur = document.querySelector("shreddit-app")), Ur)?.resolvePendingRequests();
      const e = s.headers.get("x-set-loid") || void 0,
         t = s.headers.get("x-set-session") || void 0;
      window.dispatchEvent(new CustomEvent("v2-events-sent", {
         detail: {
            loid: e,
            session_tracker: t
         }
      }))
   } 


```

They're looking for presense of `screen|view|global` in the request body.

**Reply by D4niloMR** [reply]
Now it's:
  
  ```
  if (d?.ok) {
    'basic' === d.type &&
    d.headers.has('x-verified-res') &&
    (
      (e, t = hn) => {
        if (
          e.includes('global') &&
          e.includes('view') &&
          e.includes('screen')
        ) {
          for (let n = 0; n < t.length; n++) if (
            !e.includes(t[n].field) ||
            t[n].validatePresence &&
            e.includes(`"${ t[n].field }":""`)
          ) return !1;
          return !0
        }
        return null
      }
    ) (s) &&
    (Fr || (Fr = document.querySelector('shreddit-app')), Fr) ?.resolvePendingRequests();
    const e = d.headers.get('x-set-loid') ||
    void 0,
    t = d.headers.get('x-set-session') ||
    void 0;
    window.dispatchEvent(
      new CustomEvent('v2-events-sent', {
        detail: {
          loid: e,
          session_tracker: t
        }
      })
    )
  }
  ```
  
Seems that they are checking if there's an empty value. For now it's only on `"page_type":`

**Reply by uBlock-user** [reply]
Now we need to add placholder values and cannot keep them empty

**Reply by D4niloMR** [reply]
Yes, let's hope they don't purposely break reddit for uBO users...
